source/val/validate_memory.cpp

```diff
@@ -139,6 +139,49 @@ bool HasConflictingMemberOffsets(
   return false;
 }
 
+// Returns true if |inst| contains |op| as its member or opcode of |inst| is
+// |op|.
+bool ContainsType(ValidationState_t& _, const Instruction* inst, SpvOp op) {
```

Comment:
Why are only pointers and struct subtypes checked? It seems you should check OpTypeVector, OpTypeArray, OpTypeRuntimeArray and OpTypeForwardPointer (be careful checking forward pointers).

Reply:
I updated this code to check all [*Composite* types](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#CompositeType).
I dropped OpTypePointer and OpTypeForwardPointer because the storage of value pointed by OpTypePointer or OpTypeForwardPointer may have a storage class different from the one of OpVariable.